rasa/core/actions/forms.py

```diff
@@ -355,6 +355,16 @@ def _update_slot_values(
 
         return slot_values
 
+    def _add_dynamic_slots_requested_by_dynamic_forms(
+        self, tracker: "DialogueStateTracker", domain: Domain,
+    ) -> Set[Text]:
+        required_slots = set(self.required_slots(domain))
+
+        if self.get_slot_to_fill(tracker):
+            required_slots.add(self.get_slot_to_fill(tracker))
```

Comment:
I think it is better to call the method once and reuse later on:
```
current_slots_in_form = self.get_slot_to_fill(tracker)
if current_slots_in_form:
     required_slots.add(current_slots_in_form)
```
Name `current_slots_in_form` could be better though? My context is limited.

Reply:
Agree with reusing the method.

Reply:
I'll actually label that variable `requested_slot`, we only know it's dynamic if it's not present in the `required_slots` set already.